examples/flutter_gallery/test/smoke_test.dart

```diff
@@ -43,8 +43,7 @@ Future<Null> smokeDemo(WidgetTester tester, String routeName) async {
   Finder backButton = find.byTooltip('Back');
   expect(backButton, findsOneWidget);
   await tester.tap(backButton);
-  await tester.pump(); // Start the pop "back" operation.
-  await tester.pump(const Duration(seconds: 1)); // Wait until it has finished.
+  await tester.pumpUntilNoTransientCallbacks(const Duration(seconds: 1));
```

Comment:
Scaffold's back button callback now awaits the willPop() method before actually calling Navigator.pop(). That means that tests that back out of a route by tapping the back button now need to pump three times to be consistent with the implementation. I'll add the explicit pumps (back) but binding the test to the implementation in this way doesn't seem that useful to me.